jbpm-designer-client/src/main/resources/org/jbpm/designer/public/js/documentation/docstructure.js

```diff
@@ -0,0 +1,659 @@
+            }
+        }
+
+        // properties
+        var propsArray = [];
+        var nodeName = "";
+        var elementInfoPropParts = elementInfoDetails['properties'].split(",");
+        for(var i=0; i < elementInfoPropParts.length; i++) {
+            var nextPart = elementInfoPropParts[i];
+            if(nextPart == "name") {
+                nodeName = element.properties[nextPart];
+            }
+            propsArray.push({
+                "name":presentPropertyName(nextPart),
+                "value":parsePropertyValue(nextPart,element.properties[nextPart])
+            });
+        }
+
+        myElementGroups[elementInfoDetails.group].push({
+            "id":element.resourceId,
+            "group":elementInfoDetails["group"],
+            "groupdispname":elementInfoDetails["groupdispname"],
+            "dispname":elementInfoDetails["dispname"],
+            "groupicon":ctx + elementInfoDetails["groupicon"],
+            "icon":ctx + elementInfoDetails["icon"],
+            "properties":propsArray,
+            "nodename":nodeName.trim().length > 0 ? nodeName : "(Not Named)"
+        });
+    }
+}
+
+function parsePropertyValue(propname, propvalue) {
+    if(propname == "datainputset" || propname == "dataoutputset" || propname == "datainput" || propname == "dataoutput" || propname == "vardefs") {
+        var retVal = "";
+        var propParts = propvalue.split(",");
+        for(var i=0; i < propParts.length; i++) {
+            var nextPart = propParts[i];
+            if(nextPart.indexOf(":") > 0) {
+                var innerParts = nextPart.split(":");
+                retVal += innerParts[0] +"(" + innerParts[1] + "), ";
+            } else {
+                if(nextPart.trim().length > 0) {
+                    retVal += nextPart + "(no defined type), ";
+                }
+            }
+        }
+        if(retVal.length > 0) {
+            return retVal.substring(0, retVal.length - 2);
+        } else {
+            return propvalue;
+        }
+    } else if(propname == "assignments") {
+        var retVal = "";
+
+        var valueParts = propvalue.split(",");
+        for(var i=0; i < valueParts.length; i++) {
+            var nextPart = valueParts[i];
+            if(nextPart.indexOf("=") > 0) {
+                var innerParts = nextPart.split("=");
+                if(innerParts[0].startsWith("[din]")) {
+                    var fromPart = innerParts[0].slice(5, innerParts[0].length);
+                    innerParts.shift(); // removes the first item from the array
+                    var escapedp = innerParts.join('=').replace(/\#\#/g , ",");
+                    escapedp = escapedp.replace(/\|\|/g , "=");
+
+                    retVal += "(Data Input) " + fromPart + " is given value " + escapedp + ", ";
+                } else if(innerParts[0].startsWith("[dout]")) {
+                    var fromPart = innerParts[0].slice(6, innerParts[0].length);
```

Comment:
This code is duplicated and needs to be refactored (factored out in a separate method). The only difference seems to be the first param to slice.